src/components/datePicker/DatePicker.tsx

```diff
@@ -107,7 +105,7 @@ export class DatePicker extends React.Component<IDatePickerProps, any> {
         <input
           className={inputClasses.join(' ')}
           ref={(dateInput: HTMLInputElement) => this.dateInput = dateInput}
-          onChange={() => this.handleChange()}
+          onBlur={() => this.handleChangeDate()}
```

Comment:
would be curious to see if `onChange={_.debounce(() => this.handleChangeDate(), 500)}` work here